lib/src/core/client.dart

```diff
@@ -375,10 +375,12 @@ class Web3Client {
       {EthereumAddress sender,
       @required EthereumAddress contract,
       @required Uint8List data,
+      EtherAmount value,
       BlockNum atBlock}) {
     final call = {
       'to': contract.hex,
       'data': bytesToHex(data, include0x: true, padToEvenLength: true),
+      'value': bytesToHex(value.getInWei.toRadixString(16), include0x: true),
```

Comment:
Ah wait, I think this needs to be
```suggestion
      'value': '0x${value.getInWei.toRadixString(16)}',
```

Reply:
You are right. 

Do we need to add the "value" field in "estimateGas" and "call" ?

Reply:
I think it makes sense to add them there as well, yes :+1:

Reply:
OK, let me do it.